### PR TITLE
LibWeb: Implement UserActivation & history-action activation

### PR DESCRIPTION
**Pull request by jamierocks**

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Window.h [hunk 1]
```diff
@@ -288,6 +291,9 @@ class Window final
     // https://html.spec.whatwg.org/multipage/interaction.html#user-activation-data-model
     HighResolutionTime::DOMHighResTimeStamp m_last_activation_timestamp { NumericLimits<double>::max() };
 
+    // https://html.spec.whatwg.org/multipage/interaction.html#last-history-action-activation-timestamp
+    HighResolutionTime::DOMHighResTimeStamp m_last_history_action_activation_timestamp { NumericLimits<double>::max() };
```
Comment (mattco98) [inline Userland/Libraries/LibWeb/HTML/Window.h:298]: The spec says this should be initialized to positive infinity:
```suggestion
    HighResolutionTime::DOMHighResTimeStamp m_last_history_action_activation_timestamp { Infinity<double> };
```
`NumericLimits<double>::max()` returns the highest finite number representable by `double`

Comment (jamierocks) [reply Userland/Libraries/LibWeb/HTML/Window.h]: Thank you. I hadn't realised `Infinity` existed - and just did the same thing as the code above (for last activation timestamp).

I've corrected the code I've added, and the code above (https://github.com/SerenityOS/serenity/pull/24437/commits/d1cdd5a462ff06af551b1bf1a5d398873a45f331) :)

File: Userland/Libraries/LibWeb/HTML/Navigator.h [hunk 2]
```diff
@@ -65,6 +67,9 @@ class Navigator : public Bindings::PlatformObject
 
     // https://w3c.github.io/clipboard-apis/#dom-navigator-clipboard
     JS::GCPtr<Clipboard::Clipboard> m_clipboard;
+
+    // https://html.spec.whatwg.org/multipage/interaction.html#dom-navigator-useractivation
+    JS::GCPtr<UserActivation> m_user_activation;
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/Navigator.h:72]: This needs to be visited in `Navigator::visit_edges()`.

Comment (jamierocks) [reply Userland/Libraries/LibWeb/HTML/Navigator.h:72]: Silly me - I remember thinking I must visit this when adding to the header, but never did so. Thank you :)

I've made this change.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**